common/src/main/java/org/apache/uniffle/common/netty/buffer/FileSegmentManagedBuffer.java

```diff
@@ -58,33 +60,40 @@ public ByteBuf byteBuf() {
 
   @Override
   public ByteBuffer nioByteBuffer() {
+    if (isFilled) {
+      return readByteBuffer;
+    }
     FileChannel channel = null;
     try {
       channel = new RandomAccessFile(file, "r").getChannel();
-      ByteBuffer buf = ByteBuffer.allocate(length);
+      readByteBuffer = ByteBuffer.allocate(length);
       channel.position(offset);
-      while (buf.remaining() != 0) {
-        if (channel.read(buf) == -1) {
+      while (readByteBuffer.remaining() != 0) {
+        if (channel.read(readByteBuffer) == -1) {
           throw new IOException(
               String.format(
                   "Reached EOF before filling buffer.offset=%s,file=%s,buf.remaining=%s",
-                  offset, file.getAbsoluteFile(), buf.remaining()));
+                  offset, file.getAbsoluteFile(), readByteBuffer.remaining()));
         }
       }
-      buf.flip();
-      return buf;
+      readByteBuffer.flip();
+      isFilled = true;
+      return readByteBuffer;
```

Comment:
Can we make this as readOnlyBuffer? `return buf.asReadOnlyBuffer();`

Reply:
I have tested it, but it doesn't work. `ReadOnlyBuffer` will throw a `ReadOnlyBufferException` when calling `buffer.array()`.

Reply:
Pity. I want to ensure the readBuffer will not be changed after cached.